src/confcom/azext_confcom/custom.py

```diff
@@ -163,18 +164,21 @@ def acipolicygen_confcom(
             virtual_node_yaml[count_in_file] = current_yaml
 
             os_util.write_multiple_yaml_to_file(virtual_node_yaml_path, virtual_node_yaml)
+            print(str_to_sha256(policy.get_serialized_output(OutputType.RAW, omit_id=omit_id)))
+            logger.info("CCE Policy successfully injected into YAML file")
         elif diff:
             exit_code = get_diff_outputs(policy, output_type == security_policy.OutputType.PRETTY_PRINT)
         elif arm_template and not (print_policy_to_terminal or outraw or outraw_pretty_print):
             result = inject_policy_into_template(arm_template, arm_template_parameters,
-                                                 policy.get_serialized_output(), count)
+                                                 policy.get_serialized_output(omit_id=omit_id), count)
             if result:
                 # this is always going to be the unencoded policy
-                print(str_to_sha256(policy.get_serialized_output(OutputType.RAW)))
+                print(str_to_sha256(policy.get_serialized_output(OutputType.RAW, omit_id=omit_id)))
                 logger.info("CCE Policy successfully injected into ARM Template")
         else:
             # output to terminal
-            print(f"{policy.get_serialized_output(output_type)}\n\n")
+            print(f"{policy.get_serialized_output(output_type, omit_id=omit_id)}\n\n")
+            print(str_to_sha256(policy.get_serialized_output(OutputType.RAW, omit_id=omit_id)))
```

Comment:
When `--outraw-pretty-print` is used, should the hash also be output?

Reply:
I don't think so. The workflow for setting AKV rules should use the hash of the policy injected into the ARM template. Putting it into two different steps could be confusing in my opinion

Reply:
Gotcha, asked bc I think it's being printed rn

Reply:
oh I see, it's still printing the raw version. It's probably fine this way

Reply:
kinda related, how do you view it working with diff-mode?
1. Generate policy (no omit id flag)
2. Run diff mode (with the omit id flag)

This says the policy and template match. is that ok?

Reply:
hmm I'll put this PR into draft mode and dig a bit more. I think the diff mode should only pass if the same config is used for generating the policy and the diff. But the error messaging needs to be clear on why it's failing. Thanks for catching this scenario 😊